### PR TITLE
Update src/main/java/org/getspout/spout/SpoutBlockListener.java

### PR DESCRIPTION
**Pull request by skyforce77**
Fix of duplication bug (pistons)

### CODE REVIEW

**Comment by Spouty** [comment]
**[AUTO-GENERATED COMMENT]**

Please make sure this pull request follows the [pull request guidelines](http://spout.in/prguide) in order to ensure your pull request is accepted.
- This pull request has commit(s) that are not signed off.
  
  Offending commit(s): 
- 735bc56a4e09428635a399ac102311539fcb70ca

**Comment by ZNickq** [comment]
Sign it off, and are you sure it works?

**Comment by skyforce77** [comment]
i'm sure, tested on my local server. What means "sign off"? i don't understand ^^' (i'm a github noob xD)

**Comment by thedudeguy** [comment]
add -s to your commit command. basically it just adds "Signed-off-by: [name] <[email]>" to your commit message. its a digital signature that you agree to license.

**Comment by skyforce77** [comment]
i've made this request on the website. I don't now how to on it.

File: src/main/java/org/getspout/spout/SpoutBlockListener.java [hunk 1]
```diff
@@ -127,8 +127,11 @@ private void pistonBlockMove(SpoutBlock block, BlockFace blockFace) {
 			SpoutBlock targetBlock = block.getRelative(blockFace);
 			CustomBlock material = block.getCustomBlock();
 			Byte data = block.getCustomBlockData();
+			int targetType = targetBlock.getTypeId();
+			Byte targetData = targetBlock.getData();
 			this.mm.removeBlockOverride(block);
 			this.mm.overrideBlock(targetBlock, material, data);
+			targetBlock.setTypeIdAndData(targetType, targetData, true);
```
Comment (ZNickq) [inline src/main/java/org/getspout/spout/SpoutBlockListener.java:134]: I don't see the logic in this honestly, seems like you set the id and data of the target block, after overrideBlock is called, that makes overrideBlock rather useless...

**Comment by thedudeguy** [comment]
This proposed fix doesn't work. Still fails with more than one custom block in the push, and pull event still all messed up.

**Comment by thedudeguy** [comment]
Closing this PR, since I just fixed the issue.

**Pull request closed without merging**